src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -243,6 +270,10 @@ by simp [reverse]
   (p.append q).reverse = q.reverse.append p.reverse :=
 by simp [reverse]
 
+@[simp] lemma reverse_concat {u v w : V} (p : G.walk u v) (h : G.adj v w) :
+  (p.concat h).reverse = cons (G.symm h) p.reverse :=
+by simp [concat_eq_append]
```

Comment:
Can you update `reverse_cons` to use `concat`?

Reply:
Leaving `reverse_cons` to be in terms of `append` is intentional, since I think you get more mileage out of `append` simp lemmas.

A mental model I have is that you should either work with `cons`/`append` terms or `concat`/`append` terms when you use simp, and simp shouldn't switch between these two languages. This ought to limit how many simp lemmas we need to write without needing to take the more drastic step of making `concat_eq_append` be a simp lemma.

Reply:
I guess I wasn't looking too carefully at what this lemma said when I wrote about my mental model. Maybe this shouldn't be a simp lemma for that reason...